examples/packetdump.rs

```diff
@@ -225,7 +223,33 @@ fn main() {
 
     loop {
         match rx.next() {
-            Ok(packet) => handle_packet(&interface.name[..], &EthernetPacket::new(packet).unwrap()),
+            Ok(packet) => {
+                if cfg!(any(target_os = "macos", target_os = "linux")) {
+                    if !interface.is_up() || !interface.is_running() {
+                        break;
+                    }
+                    if interface.is_broadcast(){
+                        // TAP
+                        handle_ethernet_packet(&interface.name[..], &EthernetPacket::new(packet).unwrap());
+                    }
+                    if interface.is_point_to_point(){
+                        // TUN
+                        let mut idx = 0;
+                        while packet[idx] == 0 {
+                            idx += 1;
```

Comment:
Hmm.. What is this? Is this code path triggered if used on a `tun` device? I never tried that.
Does every packet then start with a number of null bytes?

Reply:
1. yes, packet from `TUN` network device.
2. for this lib, yes, because `word aligned` (https://github.com/libpnet/libpnet/blob/master/pnet_datalink/src/bpf.rs#L79) .